src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -223,20 +228,20 @@ private void InterceptCallAndAdjustArguments(
                 },
                 extraArgument: attributeLocation);
 
-            if (!MemberSignatureComparer.InterceptorsStrictComparer.Equals(method, symbolForCompare))
+            if (!MemberSignatureComparer.InterceptorsStrictComparer.Equals(method, interceptorForCompare))
             {
                 this._diagnostics.Add(ErrorCode.WRN_InterceptorSignatureMismatch, attributeLocation, method, interceptor);
             }
 
-            method.TryGetThisParameter(out var methodThisParameter);
-            var interceptorThisParameterForCompare = needToReduce ? interceptor.Parameters[0] :
-                interceptor.TryGetThisParameter(out var interceptorThisParameter) ? interceptorThisParameter : null;
-            switch (methodThisParameter, interceptorThisParameterForCompare)
+            ParameterSymbol? methodReceiverParameter = getReceiverParameter(method, classicExtensionNotInvokedAsStatic: invokedAsExtensionMethod);
```

Comment:
>classicExtensionNotInvokedAsStatic [](http://example.com/codeflow?start=84&length=34)

I do not see any benefits from this rename. I think we should stick to the usual name ```invokedAsExtensionMethod``` that is well understood and is used throughout the codebase.